File: TsogosunProfileAdminWeb/src/app/model/loginDto/authToken.ts
import { NavItem } from "src/app/shared/menu-list-item/nav-item";
import { PrizeType } from "../profile-admin/prize-type";
import { ApplicationDto } from "./applicationDto";
import { GenderDto } from "./genderDto";
import { PointsTypeDto } from "./pointsTypeDto";
import { RankDto } from "./rankDto";
import { StatusDto } from "./statusDto";
import { UserRightsSectionDto } from "./user-rights-sectionDto";

export interface AuthToken extends UserRightsSectionDto {
    userId: number;
    userSessionId?: string | undefined;
    username?: string | undefined;
    success: boolean;
    message?: string | undefined;
    token?: string | undefined;
    applications?: ApplicationDto[] | undefined;
    navItems?: NavItem[] | undefined;
    statusItems?: StatusDto[] | undefined;
    prizeTypes?: PrizeType[] | undefined;
    ranks?: RankDto[] | undefined;
    pointsTypes?: PointsTypeDto[] | undefined;
    genders?: GenderDto[] | undefined;
    
}

File: TsogosunProfileAdmin/MSProfileAdmin/Model/Dtos/AuthToken.cs

using System;
using System.Collections.Generic;

namespace tsogosun.com.MSProfileAdmin.Model.Dtos
{
    public class AuthToken : UserRightsSectionDto
    {
        public int UserId { get; set; }
        public Guid UserSessionId { get; set; }
        public string Username { get; set; }
        public bool Success { get; set; }
        public string Message { get; set; }
        public string Token { get; set; }
        public List<ApplicationDto> Applications { get; set; }
        public List<NavItem> NavItems { get; set; }
        public List<StatusDto> StatusItems { get; set; }
        public List<PrizeType> PrizeTypes { get; set; }
        public List<RankDto> Ranks { get; set; }
        public List<PointsTypeDto> PointsTypes { get; set; }
        public List<GenderDto> Genders { get; set; }

    }
}
